samza-core/src/main/java/org/apache/samza/application/StreamApplicationDescriptorImpl.java

```diff
@@ -112,7 +112,6 @@ public StreamApplicationDescriptor withDefaultSystem(SystemDescriptor<?> default
       return expander.get().apply(this, inputDescriptor);
     }
 
-    // TODO: SAMZA-1841: need to add to the broadcast streams if inputDescriptor is for a broadcast stream
     Preconditions.checkState(!inputDescriptors.containsKey(inputDescriptor.getStreamId()),
```

Comment:
We need to add this input to broadcastStreams if the inputDescriptor.isBroadcast() is true.